PyInstaller/utils/hooks/hookutils.py

```diff
@@ -661,9 +662,111 @@ def get_pywin32_module_file_attribute(module_name):
+
+    . Spawns a subprocess importing this module and getting the value of this
+      module's `__version__` attribute.
+    . Converts both that value and the passed version string to comparable
+      tuples via the `pkg_resources.parse_version()` `setuptools` function.
+    . Returns the boolean returned by dynamically calling the private method of
+      the first such tuple corresponding to the passed comparison operator name
+      (e.g., the `tuple.__lt__()` method if that name is either `<` or `lt`),
+      passed the second such tuple.
+
+    Note that `pkg_resources.parse_version()` is generally considered to be the
+    most robust means of comparing version strings in Python. The
+    alternative `LooseVersion()` and `StrictVersion()` functions provided by the
+    standard `distutils.version` module fail for common edge-cases: e.g.,
+
+        >>> from distutils.version import LooseVersion
+        >>> LooseVersion('1.5') >= LooseVersion('1.5-rc2')
+        False
+        >>> from pkg_resources import parse_version
+        >>> parse_version('1.5') >= parse_version('1.5-rc2')
+        True
+
+    Parameters
+    ----------
+    module_name : str
+        Fully-qualified `.`-delimited module name.
+    comparison_name : str
+        Either '>=', 'ge', '<=', 'le', '>', 'gt', '<', or 'lt'.
+    module_version : str
+        PEP 0440-compliant `.`-delimited version specifier.
+
+    Returns
+    ----------
+    bool
+        Boolean returned by performing the desired module version check.
+
+    Examples
+    ----------
+        # Test whether the local version of Sphinx is 1.3.x or newer.
+        >>> from PyInstaller.utils.hooks.hookutils import is_module_version
+        >>> is_module_version('sphinx', '>=', '1.3.1')
+        True
+    """
+    # Dictionary mapping passed comparison names to private tuple method names.
+    comparison_to_method_name = {
+        '>=': '__ge__',
+        'ge': '__ge__',
+        '<=': '__le__',
+        'le': '__le__',
+        '>':  '__gt__',
+        'gt': '__gt__',
+        '<':  '__lt__',
+        'lt': '__lt__',
+    }
+
+    # If the passed comparison name is unrecognized, raise an exception.
+    if comparison_name not in comparison_to_method_name:
+        raise KeyError('Comparison name "%s" unrecognized.' % comparison_name)
+
+    # String module version obtained by importing this module in a subprocess.
+    statement = """
+import %s as module
+print(module.__version__)
+"""
+    module_version_real = exec_statement(statement % module_name)
+
+    # Convert incomparable version strings to comparable version tuples.
+    module_version_real_tuple = pkg_resources.parse_version(module_version_real)
+    module_version_fake_tuple = pkg_resources.parse_version(module_version)
+
+    # Private tuple method performing this comparison.
+    module_version_real_tuple_comparator = getattr(
+        module_version_real_tuple, comparison_to_method_name[comparison_name])
```

Comment:
>  I thought I was being clever

You **are** clever :-) Your original approach already is the important 80% of the solution. I'm just polishing to make your gold shine :-)